livingatlas/pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDValidationPipeline.java

```diff
@@ -0,0 +1,335 @@
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.apache.commons.lang.StringUtils;
+import org.gbif.dwc.terms.DwcTerm;
+import org.gbif.dwc.terms.Term;
+import org.gbif.dwc.terms.UnknownTerm;
+import org.gbif.kvs.KeyValueStore;
+import org.gbif.pipelines.ingest.options.PipelinesOptionsFactory;
+import org.gbif.pipelines.ingest.utils.FsUtils;
+import org.gbif.pipelines.io.avro.ExtendedRecord;
+import org.gbif.pipelines.parsers.utils.ModelUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.MDC;
+
+/**
+ * Pipeline responsible for validation of unique key data. This pipeline produces a YAML report that
+ * is read by {@link ALAUUIDMintingPipeline}. {@link ALAUUIDMintingPipeline} will not run if
+ * validation has failed.
+ */
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ALAUUIDValidationPipeline {
+
+  public static void main(String[] args) throws Exception {
+    String[] combinedArgs = new CombinedYamlConfiguration(args).toArgs("general", "uuid");
+    UUIDPipelineOptions options =
+        PipelinesOptionsFactory.create(UUIDPipelineOptions.class, combinedArgs);
+    MDC.put("datasetId", options.getDatasetId());
+    MDC.put("attempt", options.getAttempt().toString());
+    MDC.put("step", "VALIDATE_UUID");
+    PipelinesOptionsFactory.registerHdfs(options);
+    run(options);
+    // FIXME: Issue logged here: https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/105
+    System.exit(0);
+  }
+
+  public static void run(UUIDPipelineOptions options) throws Exception {
+
+    Pipeline p = Pipeline.create(options);
+
+    // deletePreviousValidation
+    deletePreviousValidation(options);
+
+    // validation results
+    PCollectionList<String> results = PCollectionList.<String>empty(p);
+
+    ALAPipelinesConfig config =
+        ALAPipelinesConfigFactory.getInstance(
+                options.getHdfsSiteConfig(), options.getCoreSiteConfig(), options.getProperties())
+            .get();
+
+    // create key value store for data resource metadata
+    KeyValueStore<String, ALACollectoryMetadata> dataResourceKvStore =
+        ALAAttributionKVStoreFactory.create(config);
+
+    // lookup collectory metadata for this data resource
+    ALACollectoryMetadata collectoryMetadata = dataResourceKvStore.get(options.getDatasetId());
+    if (collectoryMetadata.equals(ALACollectoryMetadata.EMPTY)) {
+      log.error("Unable to retrieve dataset metadata for dataset: " + options.getDatasetId());
+      PCollection<String> pc =
+          p.apply(Create.of(METADATA_AVAILABLE + ": false").withCoder(StringUtf8Coder.of()));
+      results = results.and(pc);
+    } else {
+      PCollection<String> pc =
+          p.apply(Create.of(METADATA_AVAILABLE + ": true").withCoder(StringUtf8Coder.of()))
+              .setCoder(StringUtf8Coder.of());
+      ;
+      results = results.and(pc);
+    }
```

Comment:
Previous code is repeated with one parameter change (true/false).

Reply:
this has been refactored to remove the duplication